### PR TITLE
[Php81] Skip param variable used in next stmt after coalesce on NewInInitializerRector

### LINKED ISSUE #8673: Incorrect behavior of NewInInitializerRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.com/demo/22a5e005-c9d2-487e-a898-78ace9769596

```php
<?php

final class DemoFile
{
    private \DateTime $time;
    private bool $timeWasSet;
    
    public function __construct(?\DateTime $time = null)
    {
        $this->time = $time ?? new \DateTime();
        $this->timeWasSet = null !== $time;
    }
}
```

Same happen with
```
$this->timeWasSet = null !== $time;
$this->time = $time ?? new \DateTime();
```

### Responsible rules

* `NewInInitializerRector`

## Expected Behavior

The code should be untouched because `$time` is used before/later.
The check `null !== $time` is false before rector changes, and true after.

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/8673

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @VincentLanglet

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

File: rules/Php81/Rector/ClassMethod/NewInInitializerRector.php [hunk 1]
```diff
@@ -112,6 +119,10 @@ public function refactor(Node $node): ?Node
                     continue;
                 }
 
+                if ($this->stmtsManipulator->isVariableUsedInNextStmt($stmts, $key + 1, $paramName)) {
```
Comment (VincentLanglet) [inline rules/Php81/Rector/ClassMethod/NewInInitializerRector.php:122]: I'm not familiar with rector code, so what does `nextstmt` means ? @samsonasik 

Does it also fix the bug for code like 
```
public function __construct(?\DateTime $time = null)
    {
        $this->timeWasSet = null !== $time; // Here the statement is before 
        $this->time = $time ?? new \DateTime();
    }
```
or
```
public function __construct(?\DateTime $time = null)
    {
        $this->time = $time ?? new \DateTime();

        // Some code...

        $a = random_int(0, 1);
        if ($a === 1) {
            $this->timeWasSet = null !== $time; // Here it's not the direct next statement
        }
    }
```

Comment (samsonasik) [reply rules/Php81/Rector/ClassMethod/NewInInitializerRector.php:122]: That mean variable used by next stmt start by key + 1 until end of stmts.

Yes, that should cover that, feel free to try latest dev-main

**Pull request merged**